sdk/state/asset_test.go

```diff
@@ -0,0 +1,34 @@
+package state_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stellar/experimental-payment-channels/sdk/state"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAsset(t *testing.T) {
+	testCases := []struct {
+		Asset             state.Asset
+		WantTxnbuildAsset txnbuild.Asset
+		WantNative        bool
```

Comment:
nit: is `WantNative` needed? If `WantTxnbuildAsset` is already checking that

Reply:
`WantNative` is the value we expect to get from the `.Native()` func, `WantTxnbuildAsset` is the value we expect to get from the `.Asset()` func.